docs/source/en/package_reference/serialization.md

```diff
@@ -6,6 +6,116 @@ rendered properly in your Markdown viewer.
+
+<Tip warning={true}>
+
+This is a very early version of the parser. The API and implementation can evolve in the near future.
+
+The parser currently does very little validation. For more details about the file format, check out https://github.com/huggingface/huggingface.js/tree/main/packages/dduf.
+
+</Tip>
+
+### How to write a DDUF file?
+
+Here is how to export a folder containing different parts of a diffusion model using [`export_folder_as_dduf`]:
+
+```python
+# Export a folder as a DDUF file
+>>> from huggingface_hub import export_folder_as_dduf
+>>> export_folder_as_dduf("FLUX.1-dev.dduf", folder_path="path/to/FLUX.1-dev")
+```
+
+For more flexibility, to can use [`export_entries_as_dduf`] and pass a list of files to include in the final DDUF file:
+
+```python
+# Export specific files from the local disk.
+>>> from huggingface_hub import export_entries_as_dduf
+>>> export_entries_as_dduf(
+...     "stable-diffusion-v1-4-FP16.dduf",
+...     entries=[ # List entries to add to the DDUF file (here, only FP16 weights)
+...         ("model_index.json", "path/to/model_index.json"),
+...         ("vae/config.json", "path/to/vae/config.json"),
+...         ("vae/diffusion_pytorch_model.fp16.safetensors", "path/to/vae/diffusion_pytorch_model.fp16.safetensors"),
+...         ("text_encoder/config.json", "path/to/text_encoder/config.json"),
+...         ("text_encoder/model.fp16.safetensors", "path/to/text_encoder/model.fp16.safetensors"),
+...         # ... add more entries here
+...     ]
+... )
+```
+
+The `entries` parameter also support passing an iterable of paths or bytes. This can prove useful if you have a loaded model and want to serialize it directly in a DDUF file instead of having to serialize each component to disk first and then as a DDUF file. Here is an example on how a `StableDiffusionPipeline` can be serialized as DDUF:
+
+
+```python
+# Export state_dicts one by one from a loaded pipeline 
+>>> from diffusers import DiffusionPipeline
+>>> from typing import Generator, Tuple
+>>> import safetensors.torch
+>>> from huggingface_hub import export_entries_as_dduf
+>>> pipe = DiffusionPipeline.from_pretrained("CompVis/stable-diffusion-v1-4")
+... # ... do some work with the pipeline
+
+>>> def as_entries(pipe: DiffusionPipeline) -> Generator[Tuple[str, bytes], None, None]:
+...     # Build an generator that yields the entries to add to the DDUF file.
+...     # The first element of the tuple is the filename in the DDUF archive (must use UNIX separator!). The second element is the content of the file.
+...     # Entries will be evaluated lazily when the DDUF file is created (only 1 entry is loaded in memory at a time)
+...     yield "vae/config.json", pipe.vae.to_json_string().encode()
```

Comment:
I was confused that this is `pipe.vae.to_json_string()` but then it's `pipe.text_encoder.config.to_json_string()` below, but I think it's correct as the vae config is a FrozenDict.

Reply:
I'd be up for another solution tbh. I just wanted to show that you need to serialize things as bytes by yourself.